### PR TITLE
#102 Onboarding Modal & Component Updates

### PR DESCRIPTION
**Pull request by brewhousedigital**
## New Onboarding Modal

For new users, when they land on the dashboard for the first time, they'll see a popup with a little carousel where we can add in content / graphics / icons. Feel free to customize this for what content you think will be useful. I left it kind of generic for now. 

image

## Homepage Updates

I noticed a behavior where the homepage would rerender the logged in view, before the instance creation widget was finished. This was causing a weird flash of content before the user was redirected to the dashboard. I realized that the entire instance generator widget was being unmounted right after it logged the user in, so instead I just added the instance generator for all users. Now if you're logged in and go to the homepage, you'll see the prompt to create a new instance. This way the "loading" state will persist even as the user is being logged into the site.

### `<AuthStateGuard>`

Several components needed the `<AuthStateGuard>` wrapper, so these were moved into the component themselves so that random pages didn't need to import it in. This way the component handles all the logic related to the component itself.

### `<InstanceGeneratorWidget>`

The Instance Creation Widget will now hide the email and password if it detects the user is logged in. It also skips the entire registration system if you are already logged in.

### `database.ts`

The instance generator function was using a mix of promises and awaits. I simplified it all down to just awaits. Did you like the promise syntax more? If so I can switch it back. Let me know what you'd like.

### `<VerifyAccountBar>`

This was causing a layout shift for about a second when the Instance Creation Widget was used. The delay between that and the redirect to the Dashboard cause the alert banner to show up. To fix this, the alert banner is now hidden until the user has dismissed the onboarding modal.

## Bug Fixes

- The `<InstanceGeneratorWidget>` component lost its `goto()` method at some point so this was added back in. Now creating an instance on the home screen will correctly send the user to the Dashboard page.
- Removed unused `fade` import on the Dashboard
- On mobile, the menu button was being pushed down because of the new version number, so the logo size is now reduced on mobile

## Miscellaneous 

Several console logs were removed across the site. They were looping and filling up the console and I couldn't read any of my own console logs lol.

### CODE REVIEW

**Comment by brewhousedigital** [comment]
This is the first pass at the onboarding modal. Let me know what you think and if you want me to change directions.

Next i'll make sure it looks fine on mobile and dark mode.

**Comment by brewhousedigital** [comment]
Dark mode and mobile looking fine for me now. Ready for review

File: packages/pockethost.io/src/util/stores.ts [hunk 1]
```diff
@@ -15,15 +15,16 @@ if (browser) {
...
   // Update derived stores when authStore changes
   authStoreState.subscribe((authStoreProps) => {
-    console.log(`subscriber change`, authStoreProps)
     isUserLoggedIn.set(authStoreProps.isValid)
     isUserVerified.set(!!authStoreProps.model?.verified)
   })
 }
+
+// Handles the intro modal for new users
+export const isFirstTimeRegistration = writable(true)
```
Comment (benallfree) [inline packages/pockethost.io/src/util/stores.ts:30]: I added a `isFirstTimeRegistrationCompleted` (default `false`) to the `users` collection. You should be able to do something now like:

```
onAuthChange( authStoreProps => {
   isFirstTimeRegistration.set(!authStoreProps.user.isFirrstTimeRegistrationCompleted)
})
```
Or something like that. You'll also probably need to add the field to the `User` record type and `authStoreProps` and add an updater to `PocketbaseClient`

**Review by benallfree — CHANGES_REQUESTED** [review]

**Comment by benallfree** [comment]
Hey @brewhousedigital just a heads up that I have a pretty big PR coming in soon. You may want to wrap this up before it gets too far away from `master`.

**Comment by brewhousedigital** [comment]
Hey @benallfree ! Holidays + work picked up real fast and threw me off my groove. Just merged your PR on this and got everything synced. 

Is setting the `isFirstTimeRegistration` value in the DB overkill? Since it's only used for like 5 seconds then discarded after the modal is dismissed?

**Comment by benallfree** [comment]
What if we introduced a `prefs` field in the users table and it was just a generic json object? Then it could be used for other things as well without having to make schema changes. Thoughts?

**Comment by brewhousedigital** [comment]
That's true! Then we could save all kinds of stuff there. Do you know of any pros/cons of using a generic JSON object vs having each preference in its own column? Might be more performant having a JSON object but it might be harder to do any analytics / filtering to see who has what preferences set.

**Comment by benallfree** [comment]
Sqlite has some pretty excellent json query abilities. I think a generic object field would solve a lot of incidental cases that aren’t worth a ton of engineering time. If you agree, I’ll add the field on the live system and then hand things back to you for finalization. How does that sound?

**Comment by brewhousedigital** [comment]
Sure, sounds good to me!

**Comment by benallfree** [comment]
Hey @brewhousedigital, check out https://github.com/benallfree/pockethost/commit/b17d9a477bd74c5cd1b82804ba0bf76c8c25c9df in `master`. I added a `preferences` field and model typings for it. As you can see, it is using `PartialDeep` to remind us that the `preferences` field is untrusted in the sense that no validation or schema is enforced at this time. PocketBase doesn't have a way to do JSON schema validation server-side, so for now we will leave it as an untrusted field where you need to manually verify that every key you use actually exists (as well as the JSON object itself).

If you like this approach, go ahead and merge. I don't think any backend changes are required at this time.

**Comment by benallfree** [comment]
Hey @brewhousedigital We are about to lose this PR because I have a big merge coming in soon. Any chance we can get this merged first?

**Comment by brewhousedigital** [comment]
Hey Ben! I've been swamped with work. If you want, we can abandon this PR and I can come back to it once your new one is merged in.

**Pull request closed without merging**